### PR TITLE
JS: Detect additional URLs in `js/incomplete-url-substring-sanitization` when the substring check is an includes check

### PR DESCRIPTION
**Pull request by erik-krogh**
Using an includes check for URLs is just generally bad. 

With this PR the `js/incomplete-url-substring-sanitization` is expanded such that URLs that contain path elements are recognized when the substring check is an includes check. 

I decided to still require that the URL starts with `http://` to make very sure that we only match URL checks.

### CODE REVIEW

File: javascript/ql/test/query-tests/Security/CWE-020/tst-IncompleteUrlSubstringSanitization.js [hunk 1]
```diff
@@ -67,5 +67,9 @@
 
 	} else {
 		doSomeThingWithTrustedURL(x);
-	}
+    }
+    
+    x.startsWith("https://secure.com/foo/bar"); // OK - the trailing slash makes prefix checks safe.
```
Comment (esbena) [inline javascript/ql/test/query-tests/Security/CWE-020/tst-IncompleteUrlSubstringSanitization.js:72]: There is no trailing slash here. Can you reformulate the explanation?

File: javascript/ql/src/Security/CWE-020/IncompleteUrlSubstringSanitization.ql [hunk 2]
```diff
@@ -43,6 +43,10 @@ where
     or
     // target is a HTTP URL to a domain on any TLD
     target.regexpMatch("(?i)https?://([a-z0-9-]+\\.)+([a-z]+)(:[0-9]+)?/?")
+    or
+    // target is a HTTP URL to a domain on any TLD with path elements, and the check is an includes check
+    check instanceof StringOps::Includes and
+    target.regexpMatch("(?i)https?://([a-z0-9-]+\\.)+([a-z]+)(:[0-9]+)?/([a-z0-9-]+/?)*")
```
Comment (esbena) [inline javascript/ql/src/Security/CWE-020/IncompleteUrlSubstringSanitization.ql:49]: Optional changes (3), pick the ones you want:

- consistency for TLDs
- more permissive paths
- flattening of quantifiers
```suggestion
    target.regexpMatch("(?i)https?://([a-z0-9-]+\\.)+" + RegExpPatterns::commonTLD() + "(:[0-9]+)?/[a-z0-9/_-]+")
```

File: change-notes/1.26/analysis-javascript.md [hunk 3]
```diff
@@ -24,6 +24,7 @@
 
 | **Query**                      | **Expected impact**          | **Change**                                                                |
 |--------------------------------|------------------------------|---------------------------------------------------------------------------|
+| Incomplete URL substring sanitization (`js/incomplete-url-substring-sanitization`) | More results | This query now recognizes additional URLs when the substring check is an includes check. |
```
Comment (esbena) [inline change-notes/1.26/analysis-javascript.md:27]:
```suggestion
| Incomplete URL substring sanitization (`js/incomplete-url-substring-sanitization`) | More results | This query now recognizes additional URLs when the substring check is an inclusion check. |
```

File: javascript/ql/src/Security/CWE-020/IncompleteUrlSubstringSanitization.ql (hunk 2 above)
Comment (erik-krogh) [reply javascript/ql/src/Security/CWE-020/IncompleteUrlSubstringSanitization.ql:49]: I'm going with 2+3.

**Pull request merged**